### PR TITLE
vmalert: update retry policy for pushing data to `-remoteWrite.url`

### PR DESCRIPTION
**Pull request by hagen1778**
By default, vmalert will make multiple retry attempts with exponential delay. The total time spent during retry attempts shouldn't exceed `-remoteWrite.retryMaxTime` (default is 30s). When retry time is exceeded vmalert drops the data dedicated for `-remoteWrite.url`. Before, vmalert dropped data after 5 retry attempts with 1s delay between attempts (not configurable).

See `-remoteWrite.retryMinInterval` and `-remoteWrite.retryMaxTime` cmd-line flags.

### CODE REVIEW

File: app/vmalert/README.md [hunk 1]
```diff
@@ -1180,6 +1180,10 @@ The shortlist of configuration flags is the following:
      Optional OAuth2 scopes to use for -notifier.url. Scopes must be delimited by ';'.
   -remoteWrite.oauth2.tokenUrl string
      Optional OAuth2 tokenURL to use for -notifier.url.
+  -remoteWrite.retryMaxTime duration
+     The max time spent on retry attempts for the failed remote-write request. Change this value if it is expected for remoteWrite.url to be unreachable for more than -remoteWrite.retryMaxTime. See also -remoteWrite.retryMinInterval (default 30s)
+  -remoteWrite.retryMinInterval duration
```
Comment (zekker6) [inline app/vmalert/README.md:1185]: doc formatting is a bit wrong here

**Review by zekker6 — APPROVED** [review]
LGTM

File: app/vmalert/README.md (hunk 1 above)
Comment (hagen1778) [reply app/vmalert/README.md:1185]: I believe it is github auto-breaking a single wide line to multiple lines...

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks!

**Pull request merged**